plugins/modules/ndo_schema_template_bd_dhcp_policy.py

```diff
@@ -0,0 +1,235 @@
+        schema=dict(type="str", required=True),
+        template=dict(type="str", required=True),
+        bd=dict(type="str", required=True),
+        dhcp_relay_policy=dict(type="str"),
+        dhcp_option_policy=dict(type="str"),
+        state=dict(type="str", default="present", choices=["absent", "present", "query"]),
+    )
+
+    module = AnsibleModule(
+        argument_spec=argument_spec,
+        supports_check_mode=True,
+        required_if=[
+            ["state", "absent", ["dhcp_relay_policy"]],
+            ["state", "present", ["dhcp_relay_policy"]],
+        ],
+    )
+
+    schema = module.params.get("schema")
+    template = module.params.get("template").replace(" ", "")
+    bd = module.params.get("bd")
+    dhcp_relay_policy = module.params.get("dhcp_relay_policy")
+    dhcp_option_policy = module.params.get("dhcp_option_policy")
+    state = module.params.get("state")
+
+    dhcp_labels_path = "/templates/{0}/bds/{1}/dhcpLabels".format(template, bd)
+
+    mso = MSOModule(module)
+
+    mso_schema = MSOSchema(mso, schema, template)
+    mso_schema.set_template_bd(bd)
+
+    tenant_id = mso_schema.schema_objects["template"].details.get("tenantId")
+
+    if dhcp_relay_policy:
+        existing_dhcp_relay_policy = {}
+        dhcp_relay_policy_uuid = get_dhcp_relay_policy_uuid(mso, tenant_id, dhcp_relay_policy)
+        mso_schema.set_template_bd_dhcp_relay_policy(dhcp_relay_policy_uuid, False)
+        if mso_schema.schema_objects.get("template_bd_dhcp_relay_policy") is not None:
+            dhcp_labels_path = "{0}/{1}".format(dhcp_labels_path, mso_schema.schema_objects["template_bd_dhcp_relay_policy"].index)
+            mso_schema.schema_objects["template_bd_dhcp_relay_policy"].details["name"] = dhcp_relay_policy
+            dhcp_option_label_ref = mso_schema.schema_objects["template_bd_dhcp_relay_policy"].details.get("dhcpOptionLabel", {}).get("ref")
+            if dhcp_option_label_ref:
+                mso_schema.schema_objects["template_bd_dhcp_relay_policy"].details["dhcpOptionLabel"]["name"] = get_dhcp_option_label_name(
+                    mso, dhcp_option_label_ref
+                )
+            existing_dhcp_relay_policy = mso_schema.schema_objects["template_bd_dhcp_relay_policy"].details
+    else:
+        existing_dhcp_relay_policy = mso_schema.schema_objects["template_bd"].details.get("dhcpLabels", [])
+        for dhcp_relay_policy in existing_dhcp_relay_policy:
+            dhcp_relay_policy["name"] = get_dhcp_relay_label_name(mso, dhcp_relay_policy.get("ref"))
+            dhcp_option_label_ref = dhcp_relay_policy.get("dhcpOptionLabel", {}).get("ref")
+            if dhcp_option_label_ref:
+                dhcp_relay_policy["dhcpOptionLabel"]["name"] = get_dhcp_option_label_name(mso, dhcp_option_label_ref)
+
+    if state == "query":
+        mso.existing = existing_dhcp_relay_policy
+        mso.exit_json()
+
+    ops = []
+    mso.previous = existing_dhcp_relay_policy
+
+    if state == "absent":
```

Comment:
```
 if state == "absent" and existing_dhcp_relay_policy:
    mso.sent = mso.existing = {}
    ops.append(dict(op="remove", path=dhcp_labels_path))
```

Reply:
ok and removed redundant empty setting of mso.sent and mso.existing